core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/credentials/parameters/aws/RoleBasedCredentialParameters.java

```diff
@@ -0,0 +1,48 @@
+package com.sequenceiq.cloudbreak.api.endpoint.v4.credentials.parameters.aws;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.credentials.parameters.Mappable;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+@ApiModel
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class RoleBasedCredentialParameters implements Mappable {
+
+    @ApiModelProperty(required = true, example = "arn:aws:iam::981628461338:role/example-role")
+    private String roleArn;
+
+    @ApiModelProperty
+    private String selector = AwsSelectorType.ROLE_BASED.getName();
```

Comment:
Enum is usable for type in JSON. Jackson is mapping automatically the strings to correct enum.